src/main/java/org/aksw/iguana/cc/worker/impl/SPARQLProtocolWorker.java

```diff
@@ -435,10 +445,22 @@ protected HttpExecutionResult buildResult() {
             // The timeout from the parameter might be reduced if the end of the time limit is near
             // and it might be so small that it causes issues.
             return future.get(config.timeout().toNanos(), TimeUnit.NANOSECONDS);
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+        } catch (InterruptedException | ExecutionException e) {
             // This will close the connection and cancel the request if it's still running.
             future.cancel(true);
             return createFailedResultBeforeRequest(queryIndex, e);
+        } catch (TimeoutException e) {
+            if (future.isDone()) {
+                LOGGER.warn("Request was already done after timeout.");
```

Comment:
I think this will not be helpful on the console. Bettwer would be:
```
Request finished immedeatly after timeout but will still be counted as timed out. 
```